### PR TITLE
nixos/redmine: Rework module

### PR DESCRIPTION
**Pull request by felixsinger**
###### Description of changes

* [nixos/redmine: Drop darcs integration](https://github.com/NixOS/nixpkgs/pull/171930/commits/3be58b2afc2369c777f987359c0843e219e78590)
* [nixos/redmine: Make optional components configurable](https://github.com/NixOS/nixpkgs/pull/171930/commits/92b1cf8b4ccc16e42b04d518f9d23de42cc97c07)
* [nixos/redmine: Add PDF export support for gant](https://github.com/NixOS/nixpkgs/pull/171930/commits/4d23eae938b33fd9cf5ad046d18fb0266b3ca8de)
* [nixos/redmine: Configure imagemagick_convert_command](https://github.com/NixOS/nixpkgs/pull/171930/commits/66ef66fc6215fa22f1f216d00ae258db93fe9441)
* [nixos/redmine: Fix PNG generation of Gant diagrams](https://github.com/NixOS/nixpkgs/pull/171930/commits/562bc5c2a9c4479b11e454d2ac75fccd13e043b9)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by 1234Brian — APPROVED** [review]

**Comment by 1234Brian** [comment]
👍

**Comment by felixsinger** [comment]
@GrahamcOfBorg test redmine.mysql redmine.pgsql

**Review by 1234Brian — APPROVED** [review]

File: nixos/modules/services/misc/redmine.nix [hunk 1]
```diff
@@ -225,16 +276,21 @@ in
...
-        scm_cvs_command = "${pkgs.cvs}/bin/cvs";
-        scm_bazaar_command = "${pkgs.breezy}/bin/bzr";
-        scm_darcs_command = "${pkgs.darcs}/bin/darcs";
+        scm_subversion_command = if cfg.components.subversion then "${pkgs.subversion}/bin/svn" else "";
+        scm_mercurial_command = if cfg.components.mercurial then "${pkgs.mercurial}/bin/hg" else "";
+        scm_git_command = if cfg.components.git then "${pkgs.git}/bin/git" else "";
+        scm_cvs_command = if cfg.components.cvs then "${pkgs.cvs}/bin/cvs" else "";
+        scm_bazaar_command = if cfg.components.breezy then "${pkgs.breezy}/bin/bzr" else "";
+        imagemagick_convert_command = if cfg.components.imagemagick then "${pkgs.imagemagick}/bin/convert" else "";
+        gs_command = if cfg.components.ghostscript then "${pkgs.ghostscript}/bin/gs" else "";
```
Comment (aanderse) [inline nixos/modules/services/misc/redmine.nix:292]: Is there a difference between providing an empty value vs not providing the key at all?

**Review by aanderse — APPROVED** [review]
Overkill for my purposes when I was managing `redmine`, but understandably useful to others... so :+1:

A minor question which probably doesn't warrant conversation, but I thought I would raise it just in case :man_shrugging:

If this is ready feel free to merge, or ping me if no one merges and I will. Thanks!

**Comment by 1234Brian** [comment]
👍 great

**Comment by winterqt** [comment]
This broke the manual build -- please switch all of the `services.redmine.components.*` documentation to use `lib.mdDoc`.

**Comment by felixsinger** [comment]
@RaitoBezarius Did you even read the comments? There are still issues open and documentation is also missing.

**Comment by aanderse** [comment]
@felixsinger let's make sure we mark such PRs as `draft` in the future.

**Comment by RaitoBezarius** [comment]
> @RaitoBezarius Did you even read the comments? There are still issues open and documentation is also missing.

I am really sorry, I am certain that I read the comments, but I did not verify the release notes as I was induced by:

> If this is ready feel free to merge, or ping me if no one merges and I will. Thanks!

and the fact there were 3 approvals.

In addition, GitHub UI does not make it very easy to review PR wich such histories.

Apologies for the early merge, I will double ask the next time in such scenarios.

**Comment by winterqt** [comment]
Would it be a good idea to revert this, @felixsinger @aanderse?

**Comment by felixsinger** [comment]
@RaitoBezarius Right, the GitHub UI makes it harder. Just keep it in mind for the future. I have seen it at different places now that people merge patches without reading the comments. Maybe the NixOS project should move to a different review system, e.g. Gerrit?

@winterqt I would say let's leave it now. I remember I completed the code and just the documentation is missing. I will to add it on the weekend if I find time.

**Comment by aanderse** [comment]
Thanks @felixsinger!

**Comment by RaitoBezarius** [comment]
> @RaitoBezarius Right, the GitHub UI makes it harder. Just keep it in mind for the future. I have seen it at different places now that people merge patches without reading the comments. Maybe the NixOS project should move to a different review system, e.g. Gerrit?

Given what happened, I will be definitely extra cautious and rather resort to double-checking, apologies again for the inconveniences to everyone. An argument that made me felt bad was this PR was running for a long time (7 months).

For Gerrit, I will defer to everyone who have Opinions™ regarding this :D.

**Pull request merged**